### DISCUSSION TITLE
Debug IConfiguration

### DISCUSSION THREAD

**Discussion by DennisJansenDev** [aws/aws-lambda-dotnet#1841] [Q&A]
Hey guys,

I am trying to debug my lambda using the lambda mock test tool for dotnet 8. Using it in rider, it was a bit of a hassle to setup, but it works now. But while debugging the application I started to notice that I can't watch my IConfiguration, it does get values out of the IConfiguration, which in turn it loads from the parameter store in AWS. So it does get the values, but I can't look into the object, it is giving me a `ArrayTypeMismatchException` when trying to fold open the IConfiguration during debugging. Even when just constructing the IConfiguration myself, before we load anything from the parameter store into it, it gives that exception. But when I run a console app without the lambda mock test tool, I can debug the IConfiguration just fine. So my gut feeling says it has something to do with the lambda mock test tool, can anyone confirm? Or does anyone have a work around for it? I am really starting to dislike AWS and its lambda's because it is so a hassle to debug and setup....

Picture of debugger for clarity:
![image](https://github.com/user-attachments/assets/e9bcecfc-deb0-477a-9b93-c132fecb7a5f)

**Comment by natanaelfiorilla** [comment]
I'm having the exact same issue. Have you found a solution?

**Comment by GarrettBeatty** [comment]
Hi, @natanaelfiorilla Can you confirm if you are using the v2 version of the test tool or the original. We have 2 tools in this directory https://github.com/aws/aws-lambda-dotnet/tree/master/Tools ?


Also are you able to give a minimal reproduction/steps for how we can replicate the issue? We have made this issue if https://github.com/aws/aws-lambda-dotnet/issues/2182 we can consolidate all updates in the issue ticket please
